test/Microsoft.AspNet.Mvc.TagHelpers.Test/Internal/FileVersionProviderTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Hosting;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Expiration.Interfaces;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class FileVersionProviderTest
+    {
+        [Fact]
+        public void AddsVersionToFiles_WhenCacheIsAbsent()
+        {
+            // Arrange
+            var appName = "testApp";
+            var filePath = "/hello/world";
+            var hostingEnvironment = GetMockHostingEnvironment(filePath);
+            var fileVersionProvider = new FileVersionProvider(hostingEnvironment.WebRootFileProvider, appName, null);
+
+            // Act
+            var result = fileVersionProvider.AddFileVersionToPath(filePath);
+
+            // Assert
+            Assert.Equal("/hello/world?v=f4OxZX_x_FO5LcGBSKHWXfwtSx-j1ncoSt3SABJtkGk", result);
+        }
+
+        [Fact]
+        public void AddsVersionToFiles_WhenAppNameIsInUrl()
```

Comment:
`WhenAppNameIsInUrl` => `WhenUrlStartsWithAppName`. 

Need to add tests for
- `/appname`
- `~/appname`
- `appname
- `foo/appname`
- `/foo/appname`
- `~/foo/appname`